documentation/modules/security/proc-recovering-expired-ca-certs.adoc

```diff
@@ -0,0 +1,88 @@
+For more information on the secrets managed in Strimzi, see xref:con-certificates-str[].
+
+NOTE: If you are using your own CA certificates, follow the procedure for xref:renewing-your-own-ca-certificates-{context}[renewing your own CA certificates].
+
+.Prerequisites
+
+* xref:deploying-cluster-operator-str[The Cluster Operator must be deployed.]
+* A Kafka cluster in which CA certificates and private keys are installed.
+* The OpenSSL TLS management tool to check the period of validity for CA certificates.
+
+In this procedure, we use a Kafka cluster named `my-cluster` within the `my-project` namespace.
+
+.Procedure
+
+. Delete the secret containing the expired CA certificate.
++
+.Deleting the Cluster CA secret
+[source,shell]
+----
+kubectl delete secret my-cluster-cluster-ca-cert -n my-project
+----
++
+.Deleting the Clients CA secret
+[source,shell]
+----
+kubectl delete secret my-cluster-clients-ca-cert -n my-project
+----
+
+. Wait for the Cluster Operator to generate new certificates. 
++
+* A new CA cluster certificate to verify the identity of the Kafka brokers is created in a secret of the same name (`my-cluster-cluster-ca-cert`).
+* A new CA clients certificate to verify the identity of Kafka users is created in a secret of the same name (`my-cluster-clients-ca-cert`).
+
+. Check the period of validity for the new CA certificates.
++
+.Checking the period of validity for the new cluster CA certificate
+[source,shell]
+----
+kubectl get secret my-cluster-cluster-ca-cert -n my-project -o=jsonpath='{.data.ca\.crt}' | base64 -d | openssl x509 -noout -dates
+----
++
+.Checking the period of validity for the new clients CA certificate
+[source,shell]
+----
+kubectl get secret my-cluster-clients-ca-cert -n my-project -o=jsonpath='{.data.ca\.crt}' | base64 -d | openssl x509 -noout -dates
+----
++
+The command returns a `notBefore` and `notAfter` date, which is the valid start and end date for the CA certificate.
+
+. Delete the component pods and secrets that use the CA certificates. 
++
+Kafka brokers, Kafka components like Kafka Connect, ZooKeeper and the Entity Operator use `my-cluster-cluster-ca-cert`.
+Kafka brokers also use `my-cluster-clients-ca-cert`.
++
+You can use the following `kubectl` command to find resources and also verify that they have been removed.
++
+[source,shell]
+----
+kubectl get <resource_type> --all-namespaces | grep <kafka_cluster_name>
+----
++
+Replace `<resource_type>` with the type of the resource, such as `Pod` or `Secret`.
+
+. Wait for the Cluster Operator to detect the missing pods and recreate them with the updated CA certificates.
+. Verify that there are no issues related to certificate validation in the Cluster Operator log.
+. Update client configurations to trust the new cluster CA certificate.
++
+See:
++
+--
+* xref:configuring-internal-clients-to-trust-cluster-ca-str[]
+* xref:configuring-external-clients-to-trust-cluster-ca-str[]
+--
```

Comment:
The things IMHO need to be deleted in the right order:

1) Delete the ZooKeeper Secret
2) Wait for the ZooKeeper secret to be renewed / recreated
3) Delete all ZooKeeper pods with `kubectl`

4) Delete the Kafka secret
5) Wait for it to be renewed / recreated
6) Delete the Kafka pods

I think EO, KE, and CC should roll automatically afterward. But not 100% sure, we would need to test it.

----

For Clients CA, only Kafka brokers and EO would matter.

Reply:
I've updated. But I'm assuming it might change when tested.